concepts/views.py

```diff
@@ -210,3 +210,74 @@ def sandbox(request, text_id):
     from annotations.models import RelationTemplate
 
     return render(request, "annotations/relationtemplate_creator.html", {})
+
+@login_required
+def add_comment_reply(request, comment_id):
```

Comment:
pretty much all these methods should check if the user has access to the concept. Just because a concept is not shown, doesn't mean one could construct a request that adds a comment to a concept (e.g. guessing the concept or comment id).